src/main/java/com/xebialabs/deployit/ci/DeployitNotifier.java

```diff
@@ -343,6 +357,16 @@ public AutoCompletionCandidates doAutoCompleteApplication(@QueryParameter final
         }
 
         public FormValidation doCheckApplication(@QueryParameter String credential, @QueryParameter final String value, @AncestorInPath AbstractProject<?, ?> project) {
```

Comment:
Might `@RequirePOST` this since it appears to open a network connection?

Reply:
implemented form validation by checking permissions